src/nupic/regions/TMRegion.hpp

```diff
@@ -112,15 +116,13 @@ class TMRegion : public RegionImpl {
     bool orColumnOutputs;
 
     // some local variables
-    UInt32 outputWidth; // columnCount *cellsPerColumn
     bool init;
-    Size iter;
+    UInt32 padding; // to prevent the next field from spanning 8 byte boundary.
```

Comment:
all these values could use UInt, why is exactly UInt32 needed?

Reply:
> all these values could use UInt, why is exactly UInt32 needed?

Because UIint is not a known width.  If something might need to be UInt64 size than we should make is so.  The execution speed of UInt32 is the same as UInt64 in these cases anyway.

Reply:
It definitely should match TM, where it's `UInt`, TM allows the change and then the region would break. 
the whole point of `typedef UInt` is abstracting the type and letting us change it. Otherwise we could have used int32_t. 
Btw, I have a test PR where I try changing UInt to some other type (unsigned int32_fast_t, or whatever, and a portion of the code breaks, but it's still possible with some changes.